File: CSharpTests/UnitTest/CalculatorTest.cs
using NUnit.Framework;
using System;

namespace UnitTest
{
    [TestFixture]
    public class CalculatorTest
    {
        private Calculator calculator;

        [SetUp]
        public void Setup()
        {
            calculator = new Calculator();
        }

        [Test]
        public void TestAddition()
        {
            Assert.That(calculator.Add(2, 3), Is.EqualTo(5));
        }

        [Test]
        public void TestSubtraction()
        {
            Assert.That(calculator.Subtract(10, 4), Is.EqualTo(6));
        }

        [Test]
        public void TestMultiplication()
        {
            Assert.That(calculator.Multiply(3, 4), Is.EqualTo(12));
        }

        [Test]
        public void TestDivision()
        {
            Assert.That(calculator.Divide(10, 2), Is.EqualTo(5));
        }

        [Test]
        public void TestDivisionByZero()
        {
            var ex = Assert.Throws<ArgumentException>(() => calculator.Divide(5, 0));
            Assert.That(ex.Message, Is.EqualTo("Cannot divide by zero"));
        }
    }
}

File: PythonTests/UnitTests/calculator_test.py
import pytest
from calculator import Calculator

calculator = Calculator()

def test_addition():
    assert calculator.add(2, 3) == 5

def test_subtraction():
    assert calculator.subtract(10, 4) == 6

def test_multiplication():
    assert calculator.multiply(3, 4) == 12

def test_division():
    assert calculator.divide(10, 2) == 5

def test_division_by_zero():
    with pytest.raises(ValueError, match="Cannot divide by zero"):
        calculator.divide(5, 0)
